torch/csrc/jit/tensorexpr/bounds_inference.h

```diff
@@ -28,11 +28,28 @@ using BoundsInfo =
 
 TORCH_API BoundsInfo inferBounds(Stmt* s, bool distinctAccessKinds = true);
 
+// Bounds inference caching the analysis. The MemDependencyChecker must already
+// have been run.
+TORCH_API BoundsInfo getInferredBounds(
+    analysis::MemDependencyChecker& analyzer,
+    Stmt* s,
+    bool distinctAccessKinds = true);
+
 TORCH_API void printBoundsInfo(const BoundsInfo& v);
 
 TORCH_API std::vector<const Expr*> getBoundExtents(
     const std::vector<TensorAccessBoundsInfo>& infos);
 
+// The kind of dependency found, in increasing order of exclusivity.
```

Comment:
If there is a WAW and RAW, which one will be reported? I think it would be better to report the RAW dependency since that is the "true" dependency.

Reply:
I see in the implementation that the order of reporting is RAW > WAR > WAW > NoDep. That looks good. 

Given that, wouldn't it be better if we reverse the order of enums here?

Reply:
Sure, I can do that.